### PR TITLE
refactor: Remove Redundancy and duplication, check for same naming pattern on Kustomize

### PR DESCRIPTION
**Pull request by jeremyharisch**
**Description**

Changes proposed in this pull request:

- Removing the term `manager` from the kustomize setup, since it is duplicated. I.e. `klm-controller-manager-role` which reads as `kyma-lifecycle-manager-controller-manager-role`, or in default profile `lifecycle-manager-manager-role`. Use only `controller` instead
- Make sure everything has `klm-` prefix in control-plane setup
- Align the resources to the following format `<prefix>-<resource type>-<details>`
    - `klm-leader-election-role` &#8594; `klm-controller-role-leader-election`
    - `klm-manager-controler-role` &#8594; `klm-controller-role` 
    - ...
- Removing manifest-role and use kubebuilder rbac generator as done for other resoruces - to get rid of roles created by ahdn use use kubebuilder framwork as much as possible
- Adapt E2E Test

<details>
  <summary>Old resources</summary>
  
  ### default
```yaml
kcp-system
---
kymas.operator.kyma-project.io
---
manifests.operator.kyma-project.io
---
moduletemplates.operator.kyma-project.io
---
watchers.operator.kyma-project.io
---
lifecycle-manager-controller-manager
---
lifecycle-manager-leader-election-role
---
lifecycle-manager-manager-role
---
lifecycle-manager-manager-role-crd
---
lifecycle-manager-manager-role-manifest
---
lifecycle-manager-metrics-reader
---
lifecycle-manager-leader-election-rolebinding
---
lifecycle-manager-manager-rolebinding
---
lifecycle-manager-manager-rolebinding-crd
---
lifecycle-manager-manager-rolebinding-manifest
---
lifecycle-manager-metrics-rolebinding
---
lifecycle-manager-metrics-service
---
lifecycle-manager-webhook-service
---
lifecycle-manager-controller-manager
---
lifecycle-manager-serving-cert
---
lifecycle-manager-selfsigned-issuer
---
lifecycle-manager-validating-webhook-configuration
```

### control-plane profile

```
kymas.operator.kyma-project.io
---
manifests.operator.kyma-project.io
---
moduletemplates.operator.kyma-project.io
---
watchers.operator.kyma-project.io
---
klm-controller-manager
---
klm-leader-election-role
---
klm-manager-role
---
klm-manager-role-crd
---
klm-manager-role-manifest
---
klm-metrics-reader
---
klm-manager-rolebinding-istio-system
---
klm-leader-election-rolebinding
---
klm-manager-rolebinding-kcp-system
---
klm-manager-rolebinding-manifest
---
klm-manager-rolebinding-kyma-system
---
klm-metrics-rolebinding
---
klm-manager-rolebinding-crd
---
klm-dashboard-mandatory-modules
---
klm-dashboard-overview
---
klm-dashboard-status
---
klm-dashboard-watcher
---
klm-event-service
---
klm-metrics-service
---
klm-webhook-service
---
klm-controller-manager
---
klm-watcher-serving-cert
---
klm-serving-cert
---
klm-watcher-selfsigned-cluster-issuer
---
klm-watcher-selfsigned-issuer
---
klm-selfsigned-issuer
---
klm-controller-manager-metrics-monitor
---
klm-watcher-gateway
---
klm-kyma-watcher
---
klm-controller-manager
---
klm-validating-webhook-configuration
```

</details>

<details>
  <summary>New Resources</summary>
  
  ### default

```
kcp-system
---
kymas.operator.kyma-project.io
---
manifests.operator.kyma-project.io
---
moduletemplates.operator.kyma-project.io
---
watchers.operator.kyma-project.io
---
lifecycle-manager-controller
---
lifecycle-manager-controller-role-leader-election
---
lifecycle-manager-controller-role
---
lifecycle-manager-controller-role-crd
---
lifecycle-manager-controller-role-metrics
---
lifecycle-manager-controller-rolebinding-leader-election
---
lifecycle-manager-controller-rolebinding-crd
---
lifecycle-manager-controller-rolebinding-metrics
---
lifecycle-manager-manager-rolebinding
---
lifecycle-manager-metrics-service
---
lifecycle-manager-webhook-service
---
lifecycle-manager-controller
---
lifecycle-manager-serving-cert
---
lifecycle-manager-selfsigned-issuer
---
lifecycle-manager-validating-webhook-configuration
```

### control plane
```
kymas.operator.kyma-project.io
---
manifests.operator.kyma-project.io
---
moduletemplates.operator.kyma-project.io
---
watchers.operator.kyma-project.io
---
klm-controller
---
klm-controller-role-leader-election
---
klm-controller-role
---
klm-controller-role-crd
---
klm-controller-role-metrics
---
klm-controller-rolebinding-istio-system
---
klm-controller-rolebinding-kcp-system
---
klm-controller-rolebinding-leader-election
---
klm-controller-rolebinding-kyma-system
---
klm-controller-rolebinding-metrics
---
klm-controller-rolebinding-crd
---
klm-dashboard-mandatory-modules
---
klm-dashboard-overview
---
klm-dashboard-status
---
klm-dashboard-watcher
---
klm-event-service
---
klm-metrics-service
---
klm-webhook-service
---
klm-controller
---
klm-watcher-serving-cert
---
klm-serving-cert
---
klm-watcher-selfsigned-cluster-issuer
---
klm-watcher-selfsigned-issuer
---
klm-selfsigned-issuer
---
klm-controller-metrics-monitor
---
klm-watcher-gateway
---
klm-kyma-watcher
---
klm-controller
---
klm-validating-webhook-configuration
```
</details>

**Related issue(s)**

### CODE REVIEW

**Comment by c-pius** [comment]
How about trying to get to something like the following:

- all KLM-related components using the `klm` prefix
- `controller-manager` is the name of the deployment (the manager wrapping our KLM controllers)
  - everything directly related to the `controller-manager` deployment includes this in the name
- all KLM components not directly related the `controller-manager` dpeloyment don't use this infix or rather one of the thing they are related to (e.g., the dashboards have no infix, watcher-related things have `watcher` infix, ...)
- rolebindings may include namespace suffix for deduplication
- resource type not used as a suffix anymore


```diff
---
kymas.operator.kyma-project.io
CustomResourceDefinition
---
manifests.operator.kyma-project.io
CustomResourceDefinition
---
moduletemplates.operator.kyma-project.io
CustomResourceDefinition
---
watchers.operator.kyma-project.io
CustomResourceDefinition
---
# the service account for the controller-manager
klm-controller-manager
ServiceAccount
---
# permissions for the controller-manager to perform leader election
- klm-leader-election-role
+ klm-controller-manager-leader-election
Role
---
# base permissions for controller-manager to perform its tasks
- klm-manager-role
+ klm-controller-manager
ClusterRole
---
# additional permissions for controller-manager to perform tasks on CRDs
- klm-manager-role-crd
+ klm-controller-manager-crd
ClusterRole
---
# additional permissions for controller-manager to read from non-resource nedpoint /metrics
- klm-metrics-reader
+ klm-controller-manager-metrics
ClusterRole
---
# binds the base permissions for istio-system namespace
- klm-manager-rolebinding-istio-system
+ klm-controller-manager-istio-system
RoleBinding
---
# binds the leader election role to kcp-sytem namespace
- klm-leader-election-rolebinding
+ klm-controller-manager-leader-election
RoleBinding
---
# binds the base permissions for kcp-system namespace
# since kcp-sysem is the expected namespace, no additional suffix?
- klm-manager-rolebinding-kcp-system
+ klm-controller-manager
RoleBinding
---
# binds the base permissions for kyma-system namespace
- klm-manager-rolebinding-kyma-system
+ klm-controller-manager-kyma-system
RoleBinding
---
# binds the metrics permissions for kcp-system namespace
- klm-metrics-rolebinding
+ klm-controller-manager-metrics
RoleBinding
---
# binds the CRD permissions
- klm-manager-rolebinding-crd
+ klm-controller-manager-crd
ClusterRoleBinding
---
# config defining a plutono dashboard
klm-dashboard-mandatory-modules
ConfigMap
---
# config defining a plutono dashboard
klm-dashboard-overview
ConfigMap
---
# config defining a plutono dashboard
klm-dashboard-status
ConfigMap
---
# config defining a plutono dashboard
klm-dashboard-watcher
ConfigMap
---
# service exposing events from the controller-manager
- klm-event-service
+ klm-controller-manager-events
Service
---
# service exposing metrics from the controller-manager
- klm-metrics-service
+ klm-controller-manager-metrics
Service
---
# service exposing the controller-manager's webhook 
- klm-webhook-service
+ klm-controller-manager-webhook
Service
---
# deployment wrapping the klm controllers
klm-controller-manager
Deployment
---
# serving cert for watcher
- klm-watcher-serving-cert
+ klm-watcher-serving
Certificate
---
# serving cert for controller-manager webhook
- klm-serving-cert
+ klm-controller-manager-webhook-serving
Certificate
---
# self signed cluster issuer for watcher
- klm-watcher-selfsigned-cluster-issuer
+ klm-watcher-selfsigned
ClusterIssuer
---
# self signed issuer for watcher
- klm-watcher-selfsigned-issuer
+ klm-watcher-selfsigned
Issuer
---
# self signed issues for controller-manager
- klm-selfsigned-issuer
+ klm-controller-manager-selfsigned
Issuer
---
# metrics monitor for controller-manager's metrics
- klm-controller-manager-metrics-monitor
+ klm-controller-manager-metrics
ServiceMonitor
---
# gateway for watcher
- klm-watcher-gateway
+ klm-watcher
Gateway
---
# watcher
- klm-kyma-watcher
+ klm-watcher
Watcher
---
klm-controller-manager
AuthorizationPolicy
---
- klm-validating-webhook-configuration
+ klm-controller-manager
ValidatingWebhookConfiguration
```

File: config/rbac/cluster_bindings/manifest_clusterrole_binding.yaml
Comment (c-pius) [inline config/rbac/cluster_bindings/manifest_clusterrole_binding.yaml:11]: not sure why this gets deleted entirely. Need to follow up on this.

Comment (c-pius) [reply config/rbac/cluster_bindings/manifest_clusterrole_binding.yaml:11]: Seems to have been moved to the base permissions. Not sure why.

Comment (c-pius) [reply config/rbac/cluster_bindings/manifest_clusterrole_binding.yaml:11]: These permissions used to be defined in a separate file. Now they have been moved to kubebuilder markers and are therefore included in the base permissions. So should be fine.

File: .github/workflows/test-smoke.yml
Comment (ruanxin) [inline .github/workflows/test-smoke.yml:131]: this should be `lifecycle-controller-manager`?

Comment (c-pius) [reply .github/workflows/test-smoke.yml:131]: True, will fix 👍🏻

Comment (c-pius) [reply .github/workflows/test-smoke.yml]: fixed here https://github.com/kyma-project/lifecycle-manager/pull/1721/commits/76758f0a88630080dc412f2bd878c0e1f71ce459

**Comment by c-pius** [comment]
Renaming ready now. We are waiting for feedback from SRE before merging.

**Review by lindnerby — APPROVED** [review]

**Pull request merged**
